### PR TITLE
fix, output option, and tests for sandbox/readstats.py

### PR DESCRIPTION
**Pull request by ctb**
@mr-c, what do you think - could this be moved to scripts/?

### CODE REVIEW

**Comment by ctb** [comment]
- [ ] Is it mergable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage.
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by mr-c** [comment]
Jenkins, retest this please

File: sandbox/readstats.py [hunk 1]
```diff
@@ -7,14 +7,22 @@
 #
 import sys
 import screed
+import argparse
 
 
 def main():
+    parser = argparse.ArgumentParser()
```
Comment (mr-c) [inline sandbox/readstats.py]: Needs a description with same usage in the epilogy to go into `scripts/`

**Comment by ctb** [comment]
jenkins, test this please

**Comment by ctb** [comment]
Ready for review @mr-c

File: tests/test_scripts.py [hunk 2]
```diff
@@ -1883,3 +1883,34 @@ def test_read_parser_streaming_bzfa():
...
     execute_load_graph_streaming(utils.get_test_data('random-20-a.fa.gz'))
+
+
+def test_readstats():
+    readstats_output = ("358 bp / 5 seqs; 71.6 average length",
+                        "916 bp / 11 seqs; 83.3 average length")
+
+    args = [utils.get_test_data("test-sweep-reads.fq"),
+            utils.get_test_data("paired-mixed.fq")]
+    status, out, err = utils.runscript('readstats.py', args, sandbox=True)
```
Comment (mr-c) [inline tests/test_scripts.py]: drop the Sandbox bit

File: tests/test_scripts.py [hunk 3]
```diff
@@ -1883,3 +1883,34 @@ def test_read_parser_streaming_bzfa():
...
+def test_readstats_output():
+    readstats_output = ("358 bp / 5 seqs; 71.6 average length",
+                        "916 bp / 11 seqs; 83.3 average length")
+
+    outfile = utils.get_temp_filename('output.txt')
+    args = ["-o", outfile,
+            utils.get_test_data("test-sweep-reads.fq"),
+            utils.get_test_data("paired-mixed.fq")]
+
+    status, _, _ = utils.runscript('readstats.py', args, sandbox=True)
```
Comment (mr-c) [inline tests/test_scripts.py]: drop the sandbox bit (keeps the test from being run)

File: scripts/readstats.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        bp = 0
+        seqs = 0
+        for record in screed.open(filename):
+            if seqs % 100000 == 0:
+                print >>sys.stderr, '...', filename, seqs
+            bp += len(record.sequence)
+            seqs += 1
+
+        if seqs == 0:
+            print >>sys.stderr, 'No sequences found in %s' % filename
```
Comment (mr-c) [inline scripts/readstats.py]: Needs test coverage

File: scripts/readstats.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                                                                avg_len,
+                                                                filename)
+            print >>sys.stderr, '... found', s
+            output.append(s)
+
+            total_bp += bp
+            total_seqs += seqs
+
+    if total_seqs == 0:
+        print >>args.outfp, \
```
Comment (mr-c) [inline scripts/readstats.py]: Needs test coverage

File: ChangeLog [hunk 6]
```diff
@@ -1,3 +1,10 @@
+2015-01-21  Titus Brown  <titus@idyll.org>
```
Comment (mr-c) [inline ChangeLog]: Update date please

**Comment by mr-c** [comment]
Fix those 5 nits and we're good.

**Comment by ctb** [comment]
@mr-c, ready for review. Looks to me like pylint picked up a few new problems because readstats was moved into scripts, and hence the CI failed; I don't think the problems are worth fixing, though. Please advise!

File: scripts/readstats.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        readstats.py tests/test-data/test-abund-read-2.fa
+    """)
+
+    parser = argparse.ArgumentParser(description=descr,
+                                     epilog=textwrap.dedent(epilog))
+    parser.add_argument('filenames', nargs='+')
+    parser.add_argument('-o', '--output', dest='outfp',
+                        help="output file for statistics; defaults to stdout.",
+                        type=argparse.FileType('w'), default=None)
+    args = parser.parse_args()
```
Comment (mr-c) [inline scripts/readstats.py]: You have to expose your Argparse object via a method `get_parser()`. This is what is failing your build

**Comment by mr-c** [comment]
We don't currently fail jobs for pylint. You can always review the log in Jenkins to see the cause of the failure.

**Comment by ctb** [comment]
Interpreting the log is always challenging; I couldn't see any clear reason why this build failed.

**Comment by mr-c** [comment]
Jenkins now knows how to recognize this failure mode

**Comment by ctb** [comment]
Jenkins, please test this.

**Comment by ctb** [comment]
Jenkins, retest this please.

**Comment by mr-c** [comment]
Well, aren't you exposing all sort of undocumented bits :-) Jenkin's Failure Cause Management has been updated to explain this one as well

**Comment by ctb** [comment]
Naturally obstreporous, that's me :)

**Comment by ctb** [comment]
OK, ready for review (& merge?) @mr-c

**Comment by mr-c** [comment]
Enjoy :-)

**Comment by ctb** [comment]
thanks!

**Pull request merged**